### ISSUE TITLE
Failed to parse speedtest.net configuration 0.3.4

### ISSUE THREAD

**Issue by vasilgerman** [sivel/speedtest-cli#271]
HI. Please, help!

$ /usr/local/bin/python2 /home/dmx/speedtest_cli.py

Retrieving speedtest.net configuration...
Failed to parse speedtest.net configuration

$ /usr/local/bin/python2 /home/dmx/speedtest_cli.py --ver
0.3.4

FreeBSD aero.cv.ua 9.3-RELEASE FreeBSD 9.3-RELEASE

Yesterday everything worked.

**Comment by comgot** [comment]
speedtest is unknown if the ISP information("speedtest-config.php line 3") sent to the Data Link Escape (U + 0010).
So, this is necessary to modify the source code.("speedtest_cli.py line 393")

[speedtest_cli.py.txt](https://github.com/sivel/speedtest-cli/files/353388/speedtest_cli.py.txt)

**Comment by galgolan** [comment]
Same issue here

**Comment by amk21** [comment]
this patch fix the problem.

there are 2 issues i had to add http header to support gzip format and i had extract the configuration  gzip 

diff --git a/speedtest_cli.py b/speedtest_cli.py
index d93d5c3..be630ca 100755
--- a/speedtest_cli.py
+++ b/speedtest_cli.py
@@ -24,6 +24,8 @@ import socket
 import timeit
 import platform
 import threading
+from StringIO import StringIO
+import gzip
 
 __version__ = '0.3.4'
 
@@ -216,6 +218,7 @@ def build_request(url, data=None, headers={}):
         schemed_url = url
 
     headers['User-Agent'] = user_agent
+    headers['Accept-Encoding'] = """gzip, deflate, sdch"""
     return Request(schemed_url, data=data, headers=headers)
 
 
@@ -382,7 +385,11 @@ def getConfig():
         sys.exit(1)
     configxml = []
     while 1:
-        configxml.append(uh.read(10240))
+        buf = StringIO(uh.read(10240))
+        f = gzip.GzipFile(fileobj=buf)
+        data = f.read()
+    
+        configxml.append(data)
         if len(configxml[-1]) == 0:
             break
     if int(uh.code) != 200:

**Comment by nodesocket** [comment]
Just encountered this problem. How can I resolve?

````bash
➜  ~ speedtest-cli
Retrieving speedtest.net configuration...
Retrieving speedtest.net server list...
Failed to parse list of speedtest.net servers
````

Running `0.3.1`.

**Comment by amk21** [comment]
apply the patch above or edit speedtest_cli.py

add the following lines after line 24 (import threading)
from StringIO import StringIO
import gzip

and 

replace line 216 
headers['User-Agent'] = user_agent

with 

headers['Accept-Encoding'] = """gzip, deflate, sdch""" return Request(schemed_url, data=data, headers=headers)

**Comment by nodesocket** [comment]
@amk21 why is the patch just pushed and a new version deployed to pip?

**Comment by amk21** [comment]
i had some issues with pushing the fix.

**Comment by sivel** [comment]
Please test with the recently released v1.0.0.
